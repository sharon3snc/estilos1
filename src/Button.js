import styled, {css} from 'styled-components'

const Button = styled.button `
    padding: 8px;
    border-radius: 4px;
    border: 1px solid black;
    background-color: black;
    color: white;

    &&:hover {
        cursor: pointer;
    }

    ${props=> props.info && css `
        background-color: white;
        color:black;
    `}

    ${props=> props.success && css `
        background-color: green;
        border-color:green;
    `}

    ${props=> props.error && css `
        background-color: red;
        border-color:red;
    `}

    ${props=> props.warning && css `
        background-color: orange;
        border-color:orange;
    `}

`

export default Button;
