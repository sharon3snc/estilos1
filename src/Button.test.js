import {render, screen} from '@testing-library/react'
import Button from "./Button"

test ('should render without error', ()=> {
    render(<Button>test</Button>)
})

test ('Should match snapshot without props', () => {
    render(<Button>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})

test ('Should match snapshot success props', () => {
    render(<Button success>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})

test ('Should match snapshot info props', () => {
    render(<Button info>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})

test ('Should match snapshot error props', () => {
    render(<Button error>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})

test ('Should match snapshot warning props', () => {
    render(<Button warning>test</Button>)
    expect(screen.getByRole('button')).toMatchSnapshot()
})