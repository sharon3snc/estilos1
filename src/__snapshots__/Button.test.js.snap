// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot error props 1`] = `
.c0 {
  padding: 8px;
  border-radius: 4px;
  border: 1px solid black;
  background-color: black;
  color: white;
  background-color: red;
  border-color: red;
}

.c0.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshot info props 1`] = `
.c0 {
  padding: 8px;
  border-radius: 4px;
  border: 1px solid black;
  background-color: black;
  color: white;
  background-color: white;
  color: black;
}

.c0.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshot success props 1`] = `
.c0 {
  padding: 8px;
  border-radius: 4px;
  border: 1px solid black;
  background-color: black;
  color: white;
  background-color: green;
  border-color: green;
}

.c0.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshot warning props 1`] = `
.c0 {
  padding: 8px;
  border-radius: 4px;
  border: 1px solid black;
  background-color: black;
  color: white;
  background-color: orange;
  border-color: orange;
}

.c0.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
>
  test
</button>
`;

exports[`Should match snapshot without props 1`] = `
.c0 {
  padding: 8px;
  border-radius: 4px;
  border: 1px solid black;
  background-color: black;
  color: white;
}

.c0.c0:hover {
  cursor: pointer;
}

<button
  class="c0"
>
  test
</button>
`;
